core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java

```diff
@@ -0,0 +1,145 @@
+package ca.jrvs.apps.grep;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.log4j.BasicConfigurator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JavaGrepImp implements JavaGrep {
+
+  final Logger logger = LoggerFactory.getLogger(JavaGrep.class);
+
+  private String regex;
+  private String rootPath;
+  private String outFile;
+
+  public JavaGrepImp(String regex, String rootPath, String outFilePath) {
+    this.regex = regex;
+    this.rootPath = rootPath;
+    this.outFile = outFilePath;
+  }
+
+  public JavaGrepImp() {
+  }
+
+  public static void main(String[] args) {
+    if (args.length != 3) {
+      throw new IllegalArgumentException("Usage: JavaGrep [regex] [rootPath] [outFilePath]");
+    }
+    JavaGrepImp grep = new JavaGrepImp(args[0], args[1], args[2]);
+    BasicConfigurator.configure(); // configure log4j
```

Comment:
wrong comment format

Reply:
you can remove line 44.

Reply:
line 44 is linked to [this issue](https://jarvis-training.slack.com/archives/CUNGTFZGQ/p1584977723026000) I had

Reply:
interestingly now I find that I don't need it right now...but I do have this issue at this first time